dask/dataframe/core.py

```diff
@@ -1134,7 +1134,41 @@ def _groupby_level0_getitem_apply(df, key, func):
     return df.groupby(level=0)[key].apply(func)
 
 
-class GroupBy(object):
+class _GroupBy(object):
+
+    def _aca_agg(self, token, func, aggfunc=None):
+        if aggfunc is None:
+            aggfunc = func
+
+        def chunk(df, index, func=func, key=self.key):
+            return func(df.groupby(index)[key])
+
+        agg = lambda df: aggfunc(df.groupby(level=0))
+        token = self._token_prefix + token
+
+        return aca([self.df, self.index], chunk=chunk, aggregate=agg,
+                   columns=self.key, token=token)
+
+    def sum(self):
+        return self._aca_agg(token='sum', func=lambda x: x.sum())
+
+    def min(self):
+        return self._aca_agg(token='min', func=lambda x: x.min())
+
+    def max(self):
+        return self._aca_agg(token='max', func=lambda x: x.max())
+
+    def count(self):
+        return self._aca_agg(token='count', func=lambda x: x.count(),
+                             aggfunc=lambda x: x.sum())
+
+    def mean(self):
+        return 1.0 * self.sum() / self.count()
```

Comment:
This is an interesting change in algorithm.  I wonder if the performance or data traversals change at all.  Assuming that the schedulers are ordering things correctly then there should be much of a difference.

Actually, with caching this would be something of an improvement because it would expose a nice set of sub-operations.